src/app/lib/business/mocking/mocking_manager.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:app/access/mocking/mocking_repository.dart';
+
+abstract interface class MockingManager {
+  factory MockingManager(MockingRepository mockingRepository) = _MockingManager;
+
+  /// Gets whether file logging is enabled.
+  bool get isMockingEnabled;
+
+  /// Gets whether mocking configuration been changed via either [setIsConsoleLoggingEnabled].
+  bool get hasConfigurationBeenChanged;
+
+  /// Sets whether mocking should be enabled on next app launch.
+  Future setIsMockingEnabled(bool isMockingEnabled);
+
+  /// Create an instancee of [Logger].
+  Future initialize();
+}
+
+/// Implementation of [MockingManager].
+/// This class is responsible for managing the mocking configuration.
+final class _MockingManager implements MockingManager {
+  final MockingRepository _mockingRepository;
+
+   late bool _initialIsMockingEnabled;
+
+  _MockingManager(this._mockingRepository);
+
+  @override
+  bool hasConfigurationBeenChanged = false;
+
+  @override
+  late bool isMockingEnabled;
+    
+  @override
+  Future initialize() async {
```

Comment:
Should we use `Future` or `Future<void>` here? I used the latter on code I've contributed in this project and to be honest I'm not sure which one is better. @Soap-141 @jeanplevesque

Reply:
I looked it up because you mentioned it, and apparently, Furure (without void) can return any time, sort of like Future<dynamic>, I guess Future<void> would be better if we know we wont return anything

Reply:
> Future<void>: It's good practice to specify the return type even if it's void for better code clarity.

From ChatGPT

Reply:
Refactor the whole application then.

Reply:
I would do it here on a separate commit, but I agree with both solutions.